### ISSUE TITLE
Gitea: dial tcp: lookup gitea-postgresql.default.svc.cluster.local:

### ISSUE THREAD

**Issue by NathanDotTo** [ubuntu/microk8s#2741]
This report is a way of me examining this gitea issue from a different perspective:

https://gitea.com/gitea/helm-chart/issues/249

```
2021/11/19 12:01:26 cmd/migrate.go:38:runMigrate() [F] Failed to initialize ORM engine: dial tcp: lookup gitea-postgresql.default.svc.cluster.local: Try again
```

Is this an application issue, or an environment issue?

I assume that gitea out-of-the-box (OOB) *should* work in a standard K8s environment. Yet gitea does not in my microk8s on Ubuntu 21.10 server VM. This leads me to *suspect* that I should be looking at the environment also.

Any clues would be appreciated, thank you. 

[inspection-report-20211119_115618.tar.gz](https://github.com/ubuntu/microk8s/files/7570120/inspection-report-20211119_115618.tar.gz)
.

**Comment by NathanDotTo** [comment]
Note that docker was also installed on the same node, but I have removed that and tested again. No difference.

**Comment by NathanDotTo** [comment]
I worked through to the point where I had the logs below, specifically:

```
cmd/migrate.go:38:runMigrate() [F] Failed to initialize ORM engine: dial tcp: lookup gitea-postgresql.default.svc.cluster.local: Try again
```

Using k cluster-info dump I saw:

```
[ERROR] plugin/errors: 2 gitea-postgresql.default.svc.cluster.local.cisco.com. A: read udp 10.1.147.194:56647->8.8.8.8:53: i/o timeout
```

That led me to test the DNS with dig and 8.8.8.8. That test didn't reveal any errors, in that DNS seemed to work. Even so, DNS seemed suspect.

So then I tried `microk8s enable storage dns:<IP address of DNS in lab>`, whereas I was previously only using `microk8s storage dns`. The `storage` part enables the persistent volumes that the database needs.

The *key* piece here is the lab DNS server IP address argument when enabling DNS with microk8s.
